o2a/mappers/action_mapper.py

```diff
@@ -24,22 +25,45 @@
 
 # pylint: disable=abstract-method
 # noinspection PyAbstractClass
-class ActionMapper(BaseMapper):
+from o2a.o2a_libs.property_utils import PropertySet
+
+
+class ActionMapper(BaseMapper, ABC):
     """Base class for all action mappers"""
 
-    def __init__(self, oozie_node: Element, name: str, trigger_rule: str = TriggerRule.ALL_SUCCESS, **kwargs):
-        super().__init__(oozie_node, name, **kwargs)
-        self.properties: Dict[str, str] = {}
-        self.trigger_rule: str = trigger_rule
+    def __init__(
+        self,
+        oozie_node: Element,
+        name: str,
+        dag_name: str,
+        job_properties: Dict[str, str],
+        configuration_properties: Dict[str, str],
+        trigger_rule: str = TriggerRule.ALL_SUCCESS,
+        **kwargs: Any,
+    ):
+        super().__init__(
+            oozie_node, name, dag_name, job_properties, configuration_properties, trigger_rule, **kwargs
+        )
+        self.action_node_properties: Dict[str, str] = {}
+        self._parse_config()
```

Comment:
I think that this should be done while parsing the node. What do you think about moving this call to `on_parse_node`?

Reply:
Yes. but only if it is done in the class and _on_parse_node uses super(). I think all logic related to config parsing should be done in the ActionMapper not in the deriving classes. I will propose a change for that.

Reply:
Please look at: https://github.com/GoogleCloudPlatform/oozie-to-airflow/pull/166/files

Reply:
Proposal here: 7e489b44fd6d74850ff3768bb5acc0b6c950b46b

Reply:
Please mark it as resolved @mik-laj  if you think my proposal is OK.